tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -11,148 +11,178 @@ using namespace shogun;
 using namespace linalg;
 using namespace Eigen;
 
-TEST(LinalgBackendEigen, SGVector_add)
-{
-	const float64_t alpha = 0.3;
-	const float64_t beta = -1.5;
+template <typename T>
+class LinalgBackendEigenAllTypesTest: public ::testing::Test { };
+template <typename T>
+class LinalgBackendEigenNonComplexTypesTest: public ::testing::Test { };
+template <typename T>
+class LinalgBackendEigenRealTypesTest: public ::testing::Test { };
+template <typename T>
+class LinalgBackendEigenNonIntegerTypesTest: public ::testing::Test { };
 
-	SGVector<float64_t> A(9);
-	SGVector<float64_t> B(9);
+// TODO: make global definitions
+// Definition of the 4 groups of Shogun types (shogun/mathematics/linalg/LinalgBackendBase.h)
+// TODO: add bool, chars and complex128_t types
+typedef ::testing::Types<int8_t, int16_t, uint16_t, int32_t, uint32_t, int64_t, uint64_t, float32_t, float64_t, floatmax_t> AllTypes;
+typedef ::testing::Types<int8_t, int16_t, uint16_t, int32_t, uint32_t, int64_t, uint64_t, float32_t, float64_t, floatmax_t> NonComplexTypes;
+typedef ::testing::Types<float32_t, float64_t, floatmax_t> RealTypes;
+// TODO: add complex128_t type
+typedef ::testing::Types<float32_t, float64_t, floatmax_t> NonIntegerTypes;
 
-	for (index_t i = 0; i < 9; ++i)
-	{
-		A[i] = i;
-		B[i] = 0.5*i;
-	}
+TYPED_TEST_CASE(LinalgBackendEigenAllTypesTest, AllTypes);
+TYPED_TEST_CASE(LinalgBackendEigenNonComplexTypesTest, NonComplexTypes);
+TYPED_TEST_CASE(LinalgBackendEigenRealTypesTest, RealTypes);
```

Comment:
@vigsterkr what's your opinion in making those things globally visible so that we wont have to re-declare them?